docs/decisions/0005-bulk-enrollment.rst

```diff
@@ -0,0 +1,59 @@
+Bulk Enrollment endpoint for Subscription learners
+--------------------------------------------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+Business need: The admin portal feature to enroll multiple learners in multiple subscription-courses
+needs a suitable backend.
+
+Various enrollment apis exist, but they do not cater to or optimize for multiple enrollments.
+
+There is an `EnrollmentApiClient` which has an `enroll_user_in_course()` method used by:
+  * models::EnterpriseCustomerUser::enroll()
+  * utils::enroll_user() ( handles multiple courses )
+  * views::CourseEnrollmentView::post()
+  * views::HandleConsentEnrollment::get()
+  * views::GrantDataSharingPermission::_enroll_learner_in_course()
+
+There is also a method to create pending user enrollments for non-edX users at:
+  `models::EnterpriseCustomer::enroll_user_pending_registration()`
+This method also handles multiple courses.
+
+Due to the scale and volume considerations, an api endpoint is needed that caters to the bulk
+use case without causing request overload.
+
+Decisions
+=========
+
+* We will add a new endpoint to edx-enterprise to handle bulk enrollment in a single request.
+* For existing edX users, we will reuse the `utils::enroll_user()` method for this use case.
+  This method handles multiple courses already. That method still ends up calling the `EnrollmentApiClient` for now.
+  But there is a cleanup effort anticipated to replace it with python api calls which is not in scope for this effort.
```

Comment:
Replace the pronouns with descriptive words so I don't have to guess what 'it' is (I only know what you mean because I am deeply involved in the project right now). Example:  "to replace the existing HttpClient calls in enroll_user() with direct calls to the enroll_user() python functions defined in the parent edx-platform Django application."